CRM/Odoosync/Mail/Error.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+/**
+ * This class sends error messages to emails from Odoo settings
+ */
+class CRM_Odoosync_Mail_Error {
+
+  /**
+   * Default domain email name
+   *
+   * @var array
+   */
+  const DEFAULT_DOMAIN_EMAIL_NAME = "CiviCRM";
+
+  /**
+   * Log messages
+   *
+   * @var array
+   */
+  private $log = [];
+
+  /**
+   * Recipients' emails use for send error messages
+   *
+   * @var array
+   */
+  private $emails = [];
+
+  /**
+   * Error message
+   *
+   * @var string
+   */
+  private $errorMessage;
+
+  /**
+   * Entity type
+   *
+   * @var string
+   */
+  private $entityType;
+
+  /**
+   * Entity id
+   *
+   * @var string
+   */
+  private $entityId;
+
+  /**
+   * Domain email name
+   *
+   * @var string
+   */
+  private $domainEmailName;
+
+  /**
+   * Domain email address
+   *
+   * @var string
+   */
+  private $domainEmailAddress;
+
+  /**
+   * CRM_Odoosync_Mail_Error constructor.
+   *
+   * @param $errorMessage
+   *
+   * @param $entityType
+   *
+   * @param $entityId
+   *
+   * @throws \CiviCRM_API3_Exception
+   * @throws \Exception
+   */
+  public function __construct($errorMessage, $entityType, $entityId) {
+    $this->setEntityId($entityId);
+    $this->setEntityType($entityType);
+    $this->setErrorMessage($errorMessage);
+    $this->setEmails();
+    $this->setLog(ts("Found %1 recipients' emails.", [ 1 => count($this->emails)]));
+
+    list($domainEmailName, $domainEmailAddress) = CRM_Core_BAO_Domain::getNameAndEmail();
+    $this->setDomainEmailAddress($domainEmailAddress);
+    $this->setDomainEmailName($domainEmailName);
+  }
+
+  /**
+   * Sends sync error message email to the recipient
+   */
+  public function sendErrorMessage(){
+    foreach ($this->emails as $email) {
+      $this->sendToEmail($email);
+    }
+  }
+
+  /**
+   * Sends sync error message email to the recipient
+   *
+   * @param $email
+   */
+  private function sendToEmail($email) {
+    $this->setLog(ts('Sending to the %1 ...', [ 1 => $email]));
```

Comment:
since the email can be sent to more than one email, it will be better to group log messages to one email togther, so in this case : 

```php
  /**
   * Sends sync error message email to the recipient
   *
   * @param $email
   */
  private function sendToEmail($email) {
    $logMessages = (ts('Sending to the %1 ...', [ 1 => $email]));
    $param = [
      'groupName' => 'msg_tpl_workflow_odoo_sync',
      'valueName' => 'civicrm_odoo_sync_error_report',
      'tplParams' =>
        [
          'errorMessage' => $this->getErrorMessage(),
          'entityType' => $this->getEntityType(),
          'entityId' => $this->getEntityId()
        ],
      'from' => $this->getDomainEmailName() . " <" . $this->getDomainEmailAddress() . ">",
      'toEmail' => $email,
    ];
    list($sent, $subject, $message, $html) = CRM_Core_BAO_MessageTemplate::sendTemplate($param);
    if ($sent === TRUE) {
      $logMessages .= ts(' => Success. Email was sent.');
    }
    else {
      $logMessages .= ts(' => Error. Email was not sent.'');
    }

 $this->setLog($logMessages);
  }

```